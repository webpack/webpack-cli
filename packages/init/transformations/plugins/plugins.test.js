"use strict";

const defineTest = require("webpack-cli-utils/defineTest");

defineTest(
	__dirname,
	"plugins",
	"plugins-0",
	[
		"new webpack.optimize.CommonsChunkPlugin({name:" +
			"'" +
			"vendor" +
			"'" +
			",filename:" +
			"'" +
			"vendor" +
			"-[hash].min.js'})"
	],
	"init"
);

defineTest(
	__dirname,
	"plugins",
	"plugins-1",
	"new webpack.optimize.DefinePlugin()",
	"add"
);

defineTest(
	__dirname,
	"plugins",
	"plugins-0",
	"new webpack.optimize.DefinePlugin()",
	"add"
);
