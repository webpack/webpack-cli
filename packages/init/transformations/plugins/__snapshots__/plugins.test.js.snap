// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins transforms correctly using "plugins-0" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    plugins: [
        new webpack.optimize.CommonsChunkPlugin({name:'vendor',filename:'vendor-[hash].min.js'})
    ]
}
"
`;

exports[`plugins transforms correctly using "plugins-0" data 2`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    plugins: [new webpack.optimize.DefinePlugin()]
}
"
`;

exports[`plugins transforms correctly using "plugins-1" data 1`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	plugins: [
        'something',
        new webpack.optimize.DefinePlugin()('/* Add your arguments here */')
    ]
}
"
`;
